dataverse-webapp/src/main/webapp/dataset.xhtml

```diff
@@ -949,25 +949,18 @@
                     <p class="text-warning">
                         <span class="glyphicon glyphicon-warning-sign"/> #{bundle['dataset.rejectMessage']}
                     </p>
-                    <ui:remove>
-                        <!--FIXME update for new comments table -->
-                        <p:inputTextarea id="returnReason" rows="3" value="#{DatasetPage.workingVersion.returnReason}"
-                                         maxlength="200" widgetVar="returnReason"
-                                         cols="70" counter="display" counterTemplate="{0} characters remaining."
-                                         autoResize="false"/>
-                        <p>
-                            <h:outputText id="display"/>
-                        </p>
-                        <p:watermark for="returnReason" value="#{bundle['dataset.rejectWatermark']}"
-                                     id="returnReasonwatermark"/>
-                    </ui:remove>
+                    <p:inputTextarea id="returnReason" rows="3" value="#{DatasetPage.returnToAuthorReason}"
+                                     maxlength="255" widgetVar="returnReason"
+                                     cols="70" counter="display" counterTemplate="{0} characters remaining."
+                                     autoResize="false"/>
```

Comment:
shouldn't returnReason be required?

Reply:
With returnReason rquired there are issues with making OK button disabled/enabled. Instead we making it required I've decided to make button (de)activation dependant on returnReason [empty] value.
IMO it's cleaner approach that to allow user to perform invalid operation - here: trying to send modification request with empty returnReason.

Reply:
But why you want this button change disabled attribute?

Reply:
When returnReason empty buttons is disabled - user can't return to author. When not empty, button enabled - it's ok to send back for modifications.

Reply:
There should be some kind of information that field is required either way.
About enabled/disabled - why we introduce something that acts differently than everywhere in the app?

Reply:
Yeah, red asterisk should be added to indicate that the field is required.

For (dis)able logic - it's rather a proposition, but apart from 'Support' modal I'm not sure if we have modal windows with required fields.

Reply:
It is not quite true. And why we consider only modal windows? It is something that could be integrated with every form (for me it's kind of start of dynamic validation system).

Reply:
Ok, I'll remake it to use the same mechanism as is being used throughout the app.